core/src/main/java/com/google/cloud/sql/core/BaseConnectionInfoCache.java

```diff
@@ -19,62 +19,39 @@
 import com.google.cloud.sql.AuthType;
 import com.google.cloud.sql.CredentialFactory;
 import com.google.cloud.sql.IpType;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.google.common.util.concurrent.ListeningScheduledExecutorService;
-import java.io.IOException;
-import java.security.KeyPair;
 import java.util.Optional;
 import java.util.stream.Collectors;
-import javax.net.ssl.SSLSocket;
 
 /**
  * This class manages information on and creates connections to a Cloud SQL instance using the Cloud
  * SQL Admin API. The operations to retrieve information with the API are largely done
  * asynchronously, and this class should be considered threadsafe.
  */
-class DefaultConnectionInfoCache {
-  private final AccessTokenSupplier accessTokenSupplier;
-  private final CloudSqlInstanceName instanceName;
-  private final RefreshStrategy refreshStrategy;
-  private final ConnectionConfig config;
-
-  /**
-   * Initializes a new Cloud SQL instance based on the given connection name.
-   *
-   * @param config instance connection name in the format "PROJECT_ID:REGION_ID:INSTANCE_ID"
-   * @param connectionInfoRepository Service class for interacting with the Cloud SQL Admin API
-   * @param executor executor used to schedule asynchronous tasks
-   * @param keyPair public/private key pair used to authenticate connections
-   */
-  DefaultConnectionInfoCache(
-      ConnectionConfig config,
-      ConnectionInfoRepository connectionInfoRepository,
-      CredentialFactory tokenSourceFactory,
-      ListeningScheduledExecutorService executor,
-      ListenableFuture<KeyPair> keyPair,
-      long minRefreshDelayMs) {
-    this.instanceName = new CloudSqlInstanceName(config.getCloudSqlInstance());
-    this.config = config;
-
+abstract class BaseConnectionInfoCache implements ConnectionInfoCache {
```

Comment:
The base class is delegating all calls to the refresh strategy. What if we avoid introducing more indirection and just had the implementations of ConnectionInfoCache call through to their refresher?

Reply:
I removed the base class.